f600233d1b154d507b782f84604bcb12/update/sys_script_include_abc17e0edba4e918b1227ea5f39619fc.xml

```diff
@@ -7,237 +7,568 @@
+        taskOrderRecord.update();
+
+		// Create Portfolio
+        const portfolio = new GlideRecord("x_g_dis_atat_portfolio");
+        portfolio.initialize();
+        portfolio.setValue("name", portfolioName);
+        portfolio.setValue("portfolio_owner", gs.getUserID());
+        
+        // TODO: remove line below adding requesting user also as manager
+        // after servicenow issue on PROD (and ONLY on PROD) fixed where 
+        // users when only owners can't see portfolio
+        portfolio.setValue("portfolio_managers", gs.getUserID());
+        portfolio.setValue("agency", portfolioAgency);
+        
+        // set CSP column with first (?) environment name from POST body
+        const cspName = environments?.[0]?.cspName;
+            
+        const optRecord = new global.GlideQuery('x_g_dis_atat_cloud_service_provider')
+			.where('name', cspName)
+			.selectOne('sys_id');	
+        
+        if(!optRecord.isEmpty()) {
+            const cspRecord = optRecord.get();
+            gs.info(`CSP Record: ${JSON.stringify(cspRecord)}`);
+            portfolio.setValue("csp", cspRecord.sys_id);
+        }
+
+        if (acquisitionPackageRecord.sys_id) {
+            portfolio.setValue("acquisition_package", acquisitionPackageRecord.sys_id);
+        }
+        portfolio.insert();
+
+		const taskOrderRecord2 = new GlideRecord("x_g_dis_atat_task_order");
+		taskOrderRecord2.get(newTaskOrderSysId);
+		taskOrderRecord2.setValue("portfolio", portfolio.sys_id);
+		taskOrderRecord2.update();
+
+		environments.forEach((env) => this.createEnvironment(env,portfolioName,portfolio));
+
+        // Check for created CLIN(s)
+        if (clinSysIds.length > 0){
+            // Return response
+            const responseMessage = "Success: created " + clinSysIds.length + " CLIN(s)";
+            result = {
+                success: responseMessage
+            };
+
+            // Update Active Task Order
+            portfolio.setValue("active_task_order", newTaskOrderSysId);							
+            portfolio.update();
+
+            // Update Acquisition Package Status (if we have one)
+            if (acquisitionPackageRecordFound) {					
+                acquisitionPackageRecord.package_status = "TASK_ORDER_AWARDED";
+                acquisitionPackageRecord.update();
+            }
+			return result;
+        }
+        else {
+            throw new Error(JSON.stringify({
+                code: "0503",
+                message: "No CLINs created"
+            }));				
+        }
+	},
+	
+	updatePortfolio: function(portfolioSysId, taskOrderNumber) {
+		const taskOrder = this.getTaskOrderData(taskOrderNumber);
+		const portfolio = new GlideRecord("x_g_dis_atat_portfolio");
+		portfolio.get(portfolioSysId);
+		
+		if (!portfolio) {
+			throw new Error(JSON.stringify({
+				code: "0404",
+				message: "No portfolio found"
+			}));
+		}
+		
+		const taskOrderRecord = new GlideRecord("x_g_dis_atat_task_order");
```

Comment:
Oh - we have one important issue here. The Task Order Number will actually be changing, so we need to:
1. Either just assume that we're always updating the Active Task Order for a given Portfolio, in which case we just grab it from the Portfolio record. Or we parse the given TO and extract the first X characters (not sure offhand how many), which would constitute the original TO. 
2. Update the TO number in addition to PoP dates.

Reply:
changed to:
```
const taskOrderRecord = new GlideRecord("x_g_dis_atat_task_order");
taskOrderRecord.addQuery("portfolio", portfolioSysId);
taskOrderRecord.query();```

Reply:
Also added:
```
        if (taskOrderRecord.getValue("task_order_number") != taskOrderNumber) {
            taskOrderRecord.setValue("task_order_number",taskOrderNumber);
            taskOrderRecord.update();
        }
```
to update the TO number if it changed.